src/MergeContext.js

```diff
@@ -108,6 +108,82 @@ class Approval {
     }
 }
 
+class StatusCheck
+{
+    constructor(raw) {
+        assert(raw.context);
+        assert(raw.state);
+        assert(raw.target_url);
+        assert(raw.description);
+
+        this.context = raw.context;
+        this.state = raw.state;
+        this.targetUrl = raw.target_url;
+        this.description = raw.description;
+
+        this._raw = raw;
+    }
+}
+
+// Passed required status checks analysis
+class StatusResult
+{
+    // checksNumber:
```

Comment:
Please s/checksNumber/expectedStatusCount/ to clarify its meaning (this is the minimum number of tests we expect to see) and to make its suffix more consistent with other Factory code that uses `count` rather than `number` (for English reasons: They say "number of apples" but "apple count").